app/services/course/assessment/answer/programming_auto_grading_service.rb

```diff
@@ -54,34 +55,61 @@ def evaluate_package(question, package)
   #   graded.
   # @param [Course::Assessment::ProgrammingEvaluationService::Result] evaluation_result The
   #   result of evaluating the package.
-  # @param [Boolean] ignore_evaluation If set to true, the evaluation results will be ignored when
+  # @param [Array<String>] graded_test_case_types The types of test cases counted
+  # towards grade/exp calculation
+  # If set to true, the evaluation results will be ignored when
```

Comment:
Check this comment? If the type is selected, the test result should be used rather than ignored.